Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.DataProtection.Kms/KmsXmlEncryptor.cs

```diff
@@ -44,27 +47,56 @@ internal KmsXmlEncryptor(KeyManagementServiceClient kmsClient, CryptoKeyName key
         /// <inheritdoc />
         public EncryptedXmlInfo Encrypt(XElement plaintextElement)
         {
-            ByteString plainTextData;
+            // Steps:
+            // 1) Generate a local symmetric key
```

Comment:
Yay!  The algorithm looks sound, and can handle any size plaintext.

According to [the internets](https://www.google.com/search?q=does+the+initialization+vector+need+to+be+secret), the initialization vector need not be kept secret.  That may obviate the need for the KeyData.proto, but I also approve the current code.

Reply:
Yup, I figure it's simplest just to keep it all in one place.